zbus: make CONFIG_HEAP_MEM_POOL_ADD_SIZE_ZBUS configurable

embRoobin opened the pull request:
https://github.com/zephyrproject-rtos/zephyr/pull/77631 introduces default non configurable values for zbus heap. This may result in insufficient heap size for application or soc,

This change :

- introduces the option to configure the heap size, without changing the usability.
- updates the documentation to refer to automatic heap calculation configuration when using zbus

Recreation of https://github.com/zephyrproject-rtos/zephyr/pull/89012 to avoid CI failures.

rodrigopex commented on doc/services/zbus/index.rst:
```diff
@@ -936,6 +936,7 @@ Related configuration options:
   a pool for the message subscriber for a set of channels;
 * :kconfig:option:`CONFIG_ZBUS_MSG_SUBSCRIBER_NET_BUF_STATIC_DATA_SIZE` the biggest message of zbus
   channels to be transported into a message buffer;
+* :kconfig:option:`CONFIG_HEAP_MEM_POOL_ADD_SIZE_ZBUS` the reserved heap size for message buffer allocation;
```
That is not only for message subscribers but also for zbus as a whole.

rodrigopex commented on subsys/zbus/Kconfig:
```diff
@@ -79,11 +79,14 @@ config ZBUS_ASSERT_MOCK
 
 config HEAP_MEM_POOL_ADD_SIZE_ZBUS
-	int
+	int "Zbus message subscriber heap pool size."
```
The heap is not only for message subscribers. 

```suggestion
	int "Zbus requested heap pool size."
```

Curiosity only. No other subsystem in Zephyr has a prompt for adding heap size. All of them are defined internally. I see value in it, allowing the user to change that in the application configuration, since the applications will differ.

rodrigopex commented on subsys/zbus/Kconfig:
```diff
@@ -79,11 +79,14 @@ config ZBUS_ASSERT_MOCK
-
+	default 0
+	help
+	  This configures heap memory pool available for configured net buffers via ZBUS_MSG_SUBSCRIBER_NET_BUF_POOL_SIZE.
+	  Can be set, if more heap is required, or default heap size is too big for the soc.
```
Please make that more generic since zbus also uses heap for other means. You can comment on that to state the importance of this configuration.

rodrigopex reviewed (changes requested):
I saw you merged the main into your branch. You need to use rebase only. Please, apply the requested changes and let's see what the CI says.

pdgendt commented on subsys/zbus/Kconfig:
```diff
@@ -79,11 +79,17 @@ config ZBUS_ASSERT_MOCK
 
 config HEAP_MEM_POOL_ADD_SIZE_ZBUS
-	int
+	int "Zbus requested heap pool size."
```
Consistent naming

```suggestion
	int "ZBus requested heap pool size."
```

pdgendt replied:
Maybe, I only commented on the changes in this PR, perhaps another PR can be opened to apply it to all?

pdgendt commented on doc/services/zbus/index.rst:
```diff
@@ -936,6 +936,8 @@ Related configuration options:
   a pool for the message subscriber for a set of channels;
 * :kconfig:option:`CONFIG_ZBUS_MSG_SUBSCRIBER_NET_BUF_STATIC_DATA_SIZE` the biggest message of zbus
   channels to be transported into a message buffer;
+* :kconfig:option:`CONFIG_HEAP_MEM_POOL_ADD_SIZE_ZBUS` the reserved heap size for zbus in a whole
```
😅 

```suggestion
* :kconfig:option:`CONFIG_HEAP_MEM_POOL_ADD_SIZE_ZBUS` the reserved heap size for ZBus in a whole
```

pdgendt commented:
Needs a rebase on `main`

pdgendt commented:
Please squash commits that change the same kconfig text.

embRoobin committed:
zbus: add user-configurable HEAP_MEM_POOL_ADD_SIZE_ZBUS

Zbus currently does only support default heap sizes.
The configuration option allows to decrease rom-footprint for embedded
applications. Adding the option to ZBus improves its flexibility and
while not increasing its complexity.

Add default of 0 value to compensate for previous cases where
HEAP_MEM_POOL_ADD_SIZE_ZBUS was defined as "".

Consistent naming 'Zbus'/'zbus' to 'ZBus'

Signed-off-by: Matthias Plöger <matthias.ploeger@phoenixcontact.com>

embRoobin committed:
doc: update heap references according to auto heap calculation

Add and exchange global heap configuration CONFIG_HEAP_MEM_POOL_SIZE
against configuration used for auto heap calculation
CONFIG_HEAP_MEM_POOL_ADD_SIZE_ZBUS.
Add CONFIG_HEAP_MEM_POOL_ADD_SIZE_ZBUS as related configuration

Clarify that HEAP is used for zbus in a whole

Consistent naming 'Zbus'/'zbus' to 'ZBus'

Signed-off-by: Matthias Plöger <matthias.ploeger@phoenixcontact.com>
